spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -505,15 +504,15 @@ def save_csv(fname_out, level, slices, metric, metric_ratio, metric_ratio_PAM50,
     :param filename: str: input filename
     :return:
     """
-    if not os.path.isfile(fname_out):
-        with open(fname_out, 'w') as csvfile:
-            header = ['filename', 'compression_level', 'slice(I->S)', metric + '_ratio', metric + '_ratio_PAM50', metric + '_ratio_PAM50_normalized']
-            writer = csv.DictWriter(csvfile, fieldnames=header)
-            writer.writeheader()
-    with open(fname_out, 'a') as csvfile:
-        csv_writer = csv.writer(csvfile, delimiter=',')
-        line = [filename, level, slices, metric_ratio, metric_ratio_PAM50, metric_ratio_norm]
-        csv_writer.writerow(line)
+    if os.path.isfile(fname_out):
+        # Concatenate existing CSV file to the new CSV file
+        dataframe_old = pd.read_csv(fname_out)
+        dataframe = pd.concat([dataframe_old, dataframe], axis=0, ignore_index=True)
+        # Merge rows that have the same 'filename', 'compression_level', and 'slice(I->S)'
+        dataframe = dataframe.groupby(
+            [dataframe.iloc[:, 0], dataframe.iloc[:, 1], dataframe.iloc[:, 2]]
+        ).max()  # Use max for a tiebreaker (this should never happen unless the user computes the same metric twice)
+    dataframe.to_csv(fname_out, index=False)
```

Comment:
Maybe we could use `na_rep='n/a'` to have `n/a` instead of empty cells.

```suggestion
    dataframe.to_csv(fname_out, na_rep='n/a', index=False)
```

Now:

<img width="1522" alt="image" src="https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/39456460/56a8e33a-7eba-4299-8f20-00f762636a55">

With `na_rep='n/a'`:

<img width="1522" alt="image" src="https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/39456460/f2abcfcc-8b11-4072-bace-93626b23de26">